gui/storage/forms.py

```diff
@@ -130,6 +130,12 @@ def _clean_zfssize_fields(form, attrs, prefix):
     for attr in attrs:
         formfield = '%s%s' % (prefix, attr)
         match = r.match(cdata[formfield].replace(' ', ''))
+        try:
+            int(cdata[formfield])
+            continue
+        except ValueError:
+            pass
+
```

Comment:
It will test all of the rest indeed. Otherwise the other code will throw an error that you need to specify a human format like 10M, 10G, 1T, etc.

By testing if it's a pure int you know you have a valid value and can continue.